java-generator/core/src/main/java/io/fabric8/java/generator/nodes/AbstractJSONSchema2Pojo.java

```diff
@@ -37,7 +37,12 @@
 
     public abstract String getType();
 
-    public abstract GeneratorResult generateJava(CompilationUnit cu);
+    public abstract GeneratorResult generateJava();
+
+    /** Takes a string and return the corresponding package name */
+    public static String packageName(String str) {
+        return str.toLowerCase(Locale.ROOT);
```

Comment:
Should some kind of additional validation happen here in order to be sure that what is returned is compliant with Java naming conventions for packages? Or can this be ignored safely? WDYT?

Reply:
`packageName` gets called on `className`s that are already sanitized ... we should have a better story overall about mangling strings (e.g. a more principled, tested and better thought throw set of utilities), I would say that this can be a follow up PR.